Adding ssh options and flags

SarahhhhFoster opened the pull request:
I've added options and flags to the ssh connection parameters, allowing more edge cases to be addressed without overriding class values

Commit:
Adding Options

Adding an object property and logic to support ssh -o options.

Commit:
Gotta be careful with Meeses

Commit:
Adding ssh flags

Commit:
Bad syntax

I put curly braces in instead of parentheses.

Commit:
Adding documentation

Commit:
Update ConnectParams.pm

wu-lee commented:
Hi there, thanks for the patch.  I'll have a look presently and check the tests etc., I've not used this class for a long time.  It's encouraging to see evidence someone is actually using this!

Might you have any tests for your new code to protect it against regressions in the future?

SarahhhhFoster commented:
I haven't built anything to test that yet, but the example that I had in the documentation I verified using Data::Dumper, so one ought to be able to built an instance of Net::SSH::Mechanize::ConnectParams using those parameters to new and verify that the results of ssh_cmd yield the same output. The output of that options hash might be a bit fragile, being unordered, though. I'm wrestling with getting the sudo auth to work correctly, and if I can get that working for me then I can start working on unit testing if you'd like me to.

Commit:
Multiline sudo password prompt regex

In case the target machine has sudo lectures enabled. Briefly and cursorily tested.
